### PR TITLE
add ability to set IAM role on master/worker ec2 instances

### PR DESCRIPTION
**Pull request by cleaton**

### CODE REVIEW

File: core/src/main/scala/sparkdeployer/ClusterConf.scala [hunk 1]
```diff
@@ -31,6 +31,7 @@ class ClusterConf(config: Config) {
     pemFile.getAbsolutePath
   }
 
+  val IAMRole = config.as[Option[String]]("IAM-role")
```
Comment (pishen) [inline core/src/main/scala/sparkdeployer/ClusterConf.scala]: Could you change this to `val iamRole = config.as[Option[String]]("iam-role")` in order to match the naming style of this configuration file?

File: core/src/main/scala/sparkdeployer/EC2Machines.scala [hunk 2]
```diff
@@ -78,6 +78,10 @@ class EC2Machines(config: Config) extends Machines with Logging {
             .withKeyName(clusterConf.keypair)
             .withMaxCount(number)
             .withMinCount(number)
+           clusterConf.IAMRole.map(name =>
```
Comment (pishen) [inline core/src/main/scala/sparkdeployer/EC2Machines.scala]: You can achieve this by adding
```
.map(req => clusterConf.iamRole.map(name => req.withIamInstanceProfile(new IamInstanceProfileSpecification().withName(name))).getOrElse(req))
```
after line 85, just like what line 86 does.

Comment (cleaton) [inline core/src/main/scala/sparkdeployer/EC2Machines.scala]: You don't think this line becomes very long, complex and hard to read?

**Comment by cleaton** [comment]
I've made the suggested adjustments

**Comment by pishen** [comment]
Merged in release 2.1.0, thanks for the contribution!

**Pull request merged**